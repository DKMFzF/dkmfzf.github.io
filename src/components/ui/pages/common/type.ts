import { ReactNode } from 'react';

export type TCommonPageUIProps = {
	pageStyles?: string;
	children: ReactNode;
};
