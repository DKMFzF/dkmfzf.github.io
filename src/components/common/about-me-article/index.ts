export { AboutMeArticle } from './about-me-article';
