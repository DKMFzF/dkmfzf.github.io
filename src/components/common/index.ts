export * from './icon';
export * from './about-me-article';
