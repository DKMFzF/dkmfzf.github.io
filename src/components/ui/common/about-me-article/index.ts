export { AboutMeArticleUI } from './about-me-article';
