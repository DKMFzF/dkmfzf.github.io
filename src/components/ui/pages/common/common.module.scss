@use '~styles/variables' as *;

$main-padding-start: 70px;
$main-padding-end: 20px;
$main-border: 1px solid $bg-primary-color;

.main {
	position: relative;
	z-index: 0;
	display: flex;
	justify-content: center;
	max-width: $max-page-width;
	block-size: 100%;
	margin: auto;
	padding-block: $main-padding-start $main-padding-end;

	&__sectionBg {
		max-width: $max-page-width;
		margin: $margin-section;
		color: white;
		color: $bg-primary-color;
		font-family: $primary-font;
		border: none;
		border-radius: $section-border-radius;
		margin-inline: $section-outsice-margin;
	}
}
