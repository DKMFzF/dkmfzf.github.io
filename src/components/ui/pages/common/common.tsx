import { FC } from 'react';
import { TCommonPageUIProps } from './type';
import styles from './common.module.scss';

export const CommonPageUI: FC<TCommonPageUIProps> = ({
	pageStyles = '',
	children
}) => (
	<main className={styles.container}>
		<section className={`${styles.sectionBg} ${pageStyles}`}>
			{children}
		</section>
	</main>
);
