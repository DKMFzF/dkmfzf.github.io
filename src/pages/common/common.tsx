import { FC } from 'react';
import { CommonPageUI } from '@ui/index';
import { TCommonPage } from './type';

export const CommonPage: FC<TCommonPage> = ({ pageStyles, children }) => (
	<CommonPageUI pageStyles={pageStyles}>{children}</CommonPageUI>
);
