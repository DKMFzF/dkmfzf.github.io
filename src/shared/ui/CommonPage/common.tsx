import { FC } from 'react';
import { TCommonPageUIProps } from './type';
import styles from './common.module.scss';

export const CommonPage: FC<TCommonPageUIProps> = ({
	pageStyles = '',
	children
}) => (
	<main className={styles.main}>
		<section className={`${styles['main__section-bg']} ${pageStyles}`}>
			{children}
		</section>
	</main>
);
