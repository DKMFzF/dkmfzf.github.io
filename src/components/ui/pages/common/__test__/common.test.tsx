import { testRenderComponent } from '@utils-testing';
import { CommonPageUI } from '../common';

describe('[CommonPageUI]', () => {
	it('Renders common-page correctly and matches snapshot', () =>
		testRenderComponent(
			<CommonPageUI pageStyles='styles'>
				<p>page-test</p>
			</CommonPageUI>
		));
});
