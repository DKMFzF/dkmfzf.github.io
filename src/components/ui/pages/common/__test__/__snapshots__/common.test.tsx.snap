// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CommonPageUI] Renders common-page correctly and matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <section
      class="main__section-bg styles"
    >
      <p>
        page-test
      </p>
    </section>
  </main>
</DocumentFragment>
`;
